internal/kubecmd/kubecmd.go

```diff
@@ -393,6 +399,39 @@ func (k *KubeCmd) applyNodeVersion(ctx context.Context, nodeVersion updatev1alph
 	return updatedNodeVersion, err
 }
 
+func (k *KubeCmd) reconcileKubeadmConfigMap(ctx context.Context) error {
+	clusterConfiguration, kubeadmConfig, err := k.getClusterConfiguration(ctx)
+	if err != nil {
+		return fmt.Errorf("getting ClusterConfig: %w", err)
+	}
+
+	for i, v := range clusterConfiguration.APIServer.ExtraVolumes {
+		if v.Name == "konnectivity-uds" {
+			clusterConfiguration.APIServer.ExtraVolumes = slices.Delete(clusterConfiguration.APIServer.ExtraVolumes, i, i+1)
+		}
+	}
+	for i, v := range clusterConfiguration.APIServer.ExtraVolumes {
+		if v.Name == "egress-config" {
+			clusterConfiguration.APIServer.ExtraVolumes = slices.Delete(clusterConfiguration.APIServer.ExtraVolumes, i, i+1)
+		}
+	}
+	delete(clusterConfiguration.APIServer.ExtraArgs, "egress-selector-config-file")
+
+	newConfigYAML, err := yaml.Marshal(clusterConfiguration)
+	if err != nil {
+		return fmt.Errorf("marshaling ClusterConfiguration: %w", err)
+	}
+
+	kubeadmConfig.Data[constants.ClusterConfigurationKey] = string(newConfigYAML)
+	k.log.Debugf("Triggering kubeadm config update now")
+	if _, err = k.kubectl.UpdateConfigMap(ctx, kubeadmConfig); err != nil {
+		return fmt.Errorf("setting new kubeadm config: %w", err)
+	}
+
+	fmt.Fprintln(k.outWriter, "Successfully reconciled the cluster's kubeadm config")
```

Comment:
nit: only run this part if there actually were changes to the config, i.e. konnectivity resources were deleted

Reply:
There should always be changes here since this function is only called if the image is a valid upgrade (see where this is called inside `UpgradeNodeVersion`. Also, since this will land in v2.14.0 every Constellation still has the entries that we want to delete.

Reply:
On second thought this is a nice optimization for when we release patch versions for v2.14